File: app/assets/javascripts/CommunicationFromProviderToPatient.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const CommunicationFromProviderToPatientSchema = DataElementSchema({
  authorDatetime: DateTime,
  relatedTo: [String],
  negationRationale: Code,
  hqmfOid: { type: String, default: '2.16.840.1.113883.10.20.28.3.9' },
  category: { type: String, default: 'communication' },
  qdmStatus: { type: String, default: 'from_provider_to_patient' },
  qdmVersion: { type: String, default: '5.3' },
  _type: { type: String, default: 'CommunicationFromProviderToPatient' },

});

module.exports.CommunicationFromProviderToPatientSchema = CommunicationFromProviderToPatientSchema;
module.exports.CommunicationFromProviderToPatient = mongoose.model('CommunicationFromProviderToPatient', CommunicationFromProviderToPatientSchema);

File: app/models/qdm/communication_from_provider_to_patient.rb
module QDM
  # app/models/qdm/communication_from_provider_to_patient.rb
  class CommunicationFromProviderToPatient < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :authorDatetime, type: DateTime
    field :relatedTo, type: Array
    field :negationRationale, type: Mixed
    field :hqmfOid, type: String, default: '2.16.840.1.113883.10.20.28.3.9'
    field :category, type: String, default: 'communication'
    field :qdmStatus, type: String, default: 'from_provider_to_patient'
    field :qdmVersion, type: String, default: '5.3'
  end
end
